src/wp-includes/block-template-utils.php

```diff
@@ -696,7 +696,7 @@ function get_block_templates( $query = array(), $template_type = 'wp_template' )
 
 			$is_not_custom   = false === array_search(
 				wp_get_theme()->get_stylesheet() . '//' . $template_file['slug'],
-				array_column( $query_result, 'id' ),
+				wp_list_pluck( $query_result, 'id' ),
```

Comment:
Great catch! Yes, PHP 7.0.0 enhanced `array_column()` to support an array of objects and 2-dimensional arrays.

I agree with the use of Core's `wp_list_pluck()` functionality for this use case. LTGM 👍 

This PR though needs unit tests to show the issue on PHP 5.6 and then it resolved via this change. I'll add those tests but am blocking commit consideration until the tests are done and ready.

Reply:
FYI spoke with @anton-vlasenko (who's on my work team). This week, he'll build the unit tests in a new PR (which should fail on the PHP 5.6 CI jobs) and then apply this PR's change (which should then pass on the PHP 5.6 CI jobs).